loader/integration.go

```diff
@@ -55,3 +56,32 @@ func GetLatestPipelineRunForSnapshotAndScenario(adapterClient client.Client, ctx
 
 	return nil, err
 }
+
+// RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot fetches all the Integration
+// PipelineRuns associated with the given Snapshot and each of the IntegrationTestScenarios.
+// After fetching them, it removes the finalizer from the PipelineRun, and returns error if any.
+func RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot(adapterClient client.Client, logger helpers.IntegrationLogger, ctx context.Context, loader ObjectLoader, snapshot applicationapiv1alpha1.Snapshot, application applicationapiv1alpha1.Application, finalizer string) error {
+	// Get all IntegrationTestScenarios
+	integrationScenarios, err := loader.GetAllIntegrationTestScenariosForApplication(adapterClient, ctx, &application)
+	if err != nil {
+		return err
+	}
+
+	for _, scenario := range *integrationScenarios {
+		// For a given Scenario and Snapshot, get all the associated Integration PipelineRuns
+		integrationPipelineRuns, err := loader.GetAllPipelineRunsForSnapshotAndScenario(adapterClient, ctx, &snapshot, &scenario)
+		if err != nil {
+			return err
+		}
+
+		// Remove finalizer from each of the PipelineRuns
+		for _, pipelineRun := range *integrationPipelineRuns {
+			err = helpers.RemoveFinalizer(adapterClient, logger, ctx, &pipelineRun, finalizer)
+			if err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
The cyclomatic complexity of this function is high, any suggestions to improve it?

Reply:
The integration pipelineRuns should also have the [Snapshot label](https://github.com/redhat-appstudio/integration-service/blob/b3f521c408d147ecc4f8811c9112172b31c0c405/tekton/integration_pipeline.go#L57-L58), we could fetch all of them using that label instead of looping throught individual integrationTestScenarios.

Reply:
Thanks, updated the logic to use the Snapshot label to fetch the related Intg PLRs instead.